src/test/java/org/prebid/server/settings/S3ApplicationSettingsTest.java

```diff
@@ -0,0 +1,383 @@
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+@RunWith(VertxUnitRunner.class)
+public class S3ApplicationSettingsTest extends VertxTest {
+
+    private static final String BUCKET = "bucket";
+    private static final String ACCOUNTS_DIR = "accounts";
+    private static final String STORED_IMPS_DIR = "stored-imps";
+    private static final String STORED_REQUESTS_DIR = "stored-requests";
+    private static final String STORED_RESPONSES_DIR = "stored-responses";
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+    private Timeout timeout;
+
+    @Mock
+    private S3AsyncClient s3AsyncClient;
+    private Vertx vertx;
+
+    private S3ApplicationSettings s3ApplicationSettings;
+
+    @Before
+    public void setUp() {
+        vertx = Vertx.vertx();
+        s3ApplicationSettings = new S3ApplicationSettings(s3AsyncClient, BUCKET, ACCOUNTS_DIR,
+                STORED_IMPS_DIR, STORED_REQUESTS_DIR, STORED_RESPONSES_DIR, jacksonMapper, vertx);
+
+        final Clock clock = Clock.fixed(Instant.now(), ZoneId.systemDefault());
+        final TimeoutFactory timeoutFactory = new TimeoutFactory(clock);
+        timeout = timeoutFactory.create(500L);
+    }
+
+    @After
+    public void tearDown(TestContext context) {
+        vertx.close(context.asyncAssertSuccess());
+    }
+
+    @Test
+    public void getAccountByIdShouldReturnFetchedAccount(TestContext context) throws JsonProcessingException {
+        // given
+        final Account account = Account.builder()
+                .id("someId")
+                .auction(AccountAuctionConfig.builder()
+                        .priceGranularity("testPriceGranularity")
+                        .build())
+                .privacy(AccountPrivacyConfig.builder().build())
+                .build();
+
+        given(s3AsyncClient.getObject(any(GetObjectRequest.class), any(AsyncResponseTransformer.class)))
+                .willReturn(CompletableFuture.completedFuture(
+                        ResponseBytes.fromByteArray(
+                                GetObjectResponse.builder().build(),
+                                mapper.writeValueAsString(account).getBytes())));
+
+        // when
+        final Future<Account> future = s3ApplicationSettings.getAccountById("someId", timeout);
+
+        // then
+        final Async async = context.async();
+
+        future.onComplete(context.asyncAssertSuccess(returnedAccount -> {
+            assertThat(returnedAccount.getId()).isEqualTo("someId");
+            assertThat(returnedAccount.getAuction().getPriceGranularity()).isEqualTo("testPriceGranularity");
+
+            verify(s3AsyncClient).getObject(
+                    eq(GetObjectRequest.builder().bucket(BUCKET).key(ACCOUNTS_DIR + "/someId.json").build()),
+                    any(AsyncResponseTransformer.class));
+            async.complete();
+        }));
+    }
```

Comment:
here is an example of assertions re-writing for junit5
```java
future.onComplete(context.succeeding(result -> {
            // place the required assertions or mockito.verify() here
            ...
            context.completeNow(); // it's mandatory to close the context
}));
```

P.S. for the future failing case use the following
```java
future.onComplete(context.failing(result -> {
            // place the required assertions or mockito.verify() here
            ...
            context.completeNow(); // it's mandatory to close the context
}));
```

Reply:
Thanks a lot :heart:  This was super helpful